gateway/ln-gateway/src/lnd.rs

```diff
@@ -318,17 +321,38 @@ impl ILnRpcClient for GatewayLndClient {
         Ok(GetRouteHintsResponse { route_hints })
     }
 
-    async fn pay(&self, invoice: PayInvoiceRequest) -> crate::Result<PayInvoiceResponse> {
+    async fn pay(&self, request: PayInvoiceRequest) -> crate::Result<PayInvoiceResponse> {
+        let PayInvoiceRequest {
+            invoice,
+            max_fee_msat,
+            ..
+        } = request;
+
         let mut client = Self::connect(
             self.address.clone(),
             self.tls_cert.clone(),
             self.macaroon.clone(),
         )
         .await?;
+
+        let fee_cap: Result<i64, _> = max_fee_msat.try_into();
+        let fee_limit = match fee_cap {
+            Ok(fee_cap) => Some(FeeLimit {
+                limit: Some(Limit::FixedMsat(fee_cap)),
+            }),
+            // LND API allows fee limits in the `i64` range, but we use `u64` for max_fee_msat.
+            // This means we can only set an enforceable fee limit between 0 and i64::MAX
+            //
+            // In case the value we try to set is too large for the range allowed by LND API
+            // Avoid setting a fee limit and let LND use its defaults
+            Err(_) => None,
```

Comment:
The safe thing to do would be to abort the payment instead of proceeding without a limit.